### PR TITLE
Expand HTTP retry/rate-limit test coverage and add sandbox diagnostics page

### LINKED ISSUE #10: [Test] Increase HTTP client test coverage for retries, rate limits, and failure modes
## Summary
Expand test coverage around HTTP pipeline behavior to reduce regression risk as MVP features are added.

## Why
Current tests validate basic fundamentals but do not deeply exercise retry/rate-limit/transport edge cases.

## Target scenarios
- Retry behavior across transient failures.
- 429 handling with and without `Retry-After`.
- 5xx handling and retry exhaustion.
- Non-retriable errors.
- Token refresh timing and concurrency behavior.
- Cancellation behavior while delaying/retrying.

## Acceptance criteria
- [ ] Unit tests added for core HTTP decision paths.
- [ ] At least one higher-level integration-style test with custom message handler.
- [ ] Flaky behavior avoided; tests deterministic and CI-stable.
- [ ] Coverage improvements measurable in changed HTTP/auth components.

## Goal links
- G4 (safe and stable production usage)
- G2 (predictable developer experience)

## Scope notes
Focus on test depth and reliability; avoid broad endpoint expansion in this issue.

### PR DESCRIPTION
**Pull request by markheydon**
Closes #10

## Summary

Two related tracks of work to improve HTTP client reliability visibility and test confidence.

### Track 1 — SDK: per-request headers + expanded retry test coverage

**Changes to `FreeAgentHttpClient`:**

- Added a per-request header seam: all HTTP method helpers (`GetAsync`, `PostAsync`, `PatchAsync`, `DeleteAsync`) now accept an optional `IEnumerable<KeyValuePair<string, string>>?` parameter. Existing call sites are unaffected (defaults to `null`).
- Added `ValidateRequestHeaders()` which throws `InvalidOperationException` if `X-RateLimit-Test` is included on a non-sandbox environment, preventing the header from reaching production accidentally.
- Refactored the retry loop to use a `createRequest` factory lambda calling `CreateRequestMessage()` on every attempt, so headers are correctly attached to each fresh `HttpRequestMessage`.
- Added a testability constructor accepting `HttpClient`, `FreeAgentOAuthClient`, and `OAuthTokenResponse` to enable deterministic unit tests without DI container setup.

**16 new tests in `FreeAgentHttpClientRetryTests`:**

| Test | Scenario |
|---|---|
| `RetryOnServerError_EventuallySucceeds` | 5xx retry until success |
| `RetryOnNetworkFailure_EventuallySucceeds` | `HttpRequestException` retry |
| `RetryOn429_RespectsRetryAfterDeltaSeconds` | Retry-After: N seconds |
| `RetryOn429_FallsBackTo60SecondsWhenNoRetryAfter` | Missing Retry-After header |
| `RetryOn429_RespectsRetryAfterHttpDate` | Retry-After: HTTP date |
| `ServerErrorExhaustsRetries_ThrowsFreeAgentException` | 5xx exhaustion with attempt metadata |
| `NoRetryOn4xxClientError` | Non-retriable 404 |
| `CancellationDuringRetryDelay_ThrowsOperationCancelled` | Cancel during back-off |
| `CancellationDuringRateLimitDelay_ThrowsOperationCancelled` | Cancel during 429 wait |
| `PerRequestHeaders_SentOnEveryRetryAttempt` | Headers preserved across retries |
| `XRateLimitTestHeader_AllowedInSandbox` | Sandbox guard: allow |
| `XRateLimitTestHeader_RejectedOutsideSandbox` | Sandbox guard: reject |
| `TokenRefresh_TriggeredWhenExpiringWithinThreshold` | Token refresh timing |
| `TokenRefresh_OnlyRefreshedOnceUnderConcurrentRequests` | Concurrency: single refresh |
| `TokenRefresh_RefreshedTokenUsedForSubsequentRequests` | Refreshed token in Authorization header |
| `CustomHandler_IntegrationStyle_ReturnsDeserialisedResponse` | Integration-style handler wiring |

### Track 2 — Sample app: Sandbox Rate-Limit Diagnostics page

New page at `/sandbox/rate-limit-test` (`SandboxRateLimitTest.razor`):

- Configurable request count (1–25).
- Toggle to send `X-RateLimit-Test: true` (lowers sandbox rate limit to 5 req/min for testing).
- Sandbox-only guard — disabled with an error message outside sandbox.
- Results table: Attempt / Timestamp / Outcome / Retry-After / X-RateLimit-Limit / X-RateLimit-Remaining / X-RateLimit-Reset / Details.
- Uses `MaxNetworkRetries=0` so raw 429 responses surface directly.

"Developer Tools" nav group added to `MainLayout.razor` and `samples/README.md` updated with usage notes.

## Test results

```
Test Run Successful.
Total tests: 59 — Passed: 59 — Failed: 0 — Skipped: 0
```

## Checklist

- [x] All existing tests pass
- [x] 16 new tests added and passing
- [x] Sample app builds successfully
- [x] `X-RateLimit-Test` blocked outside sandbox
- [x] Per-request headers preserved on every retry attempt
- [x] `samples/README.md` updated
- [x] No new NuGet packages added

### CODE REVIEW

File: src/FreeAgent.Client/Infrastructure/Http/FreeAgentHttpClient.cs [hunk 1]
```diff
@@ -95,10 +100,46 @@ public FreeAgentHttpClient(HttpClient httpClient, string accessToken, FreeAgentH
...
+        _oauthClient = oauthClient ?? throw new ArgumentNullException(nameof(oauthClient));
+        _currentToken = token ?? throw new ArgumentNullException(nameof(token));
+        _ownsHttpClient = false;
+        _options = options ?? new FreeAgentHttpClientOptions();
+        _environment = environment;
+
+        if (_httpClient.BaseAddress == null)
+        {
+            _httpClient.BaseAddress = new Uri(FreeAgentEnvironmentEndpoints.GetApiBaseUrl(environment));
+        }
```
[the remark this answers is not in the document]
Comment (markheydon) [reply src/FreeAgent.Client/Infrastructure/Http/FreeAgentHttpClient.cs]: Fixed. When `BaseAddress` is already set on the injected `HttpClient`, `_environment` is now derived from it (checking for `api.sandbox.freeagent.com`) instead of blindly trusting the `environment` parameter — identical to the behaviour of the main constructor. If `BaseAddress` is null, the provided `environment` is used to set it and `_environment` is assigned from the parameter as before.
